packages/core/src/components/forms/numericInput.tsx

```diff
@@ -524,6 +524,23 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
     private isFloatingPointNumericCharacter(char: string) {
         return NumericInput.FLOATING_POINT_NUMBER_CHARACTER_REGEX.test(char);
     }
+
+    private getStepMaxPrecision(props: HTMLInputProps & INumericInputProps) {
+        let stepMaxPrecision = Utils.countDecimalPlaces(props.stepSize);
+        if (props.minorStepSize != null) {
+            stepMaxPrecision = Math.max(stepMaxPrecision, Utils.countDecimalPlaces(props.minorStepSize));
+        }
+
+        return stepMaxPrecision;
+    }
+
+    private toMaxPrecision(value: number) {
+        const { stepMaxPrecision } = this.state;
+        // round the value to have the specified maximum precision (toFixed is the wrong choice,
+        // because it would show trailing zeros in the decimal part out to the specified precision)
+        // source: http://stackoverflow.com/a/18358056/5199574
+        return parseFloat(Math.round(parseFloat(value + "e+" + stepMaxPrecision)) + "e-" + stepMaxPrecision);
```

Comment:
Voila: https://jsperf.com/roundtomaxprecision. Will tweak this line based on these findings.